searches/linear_search.py

```diff
@@ -38,6 +38,39 @@ def linear_search(sequence, target):
     return None
 
 
+def rec_linear_search(sequence, low, high, target) -> int:
+    '''
+    Pure implementation of recursive linear search algorithm in Python
+
+    :param sequence: a collection with comparable items (as sorted items not required
+        in Linear Search)
+    :param low: Lower bound of the sequence
+    :param high: Higher bound of the sequence
+    :param target: The element to be found
+    :return: Index of the target or None if target not found
+
+    Examples:
+    >>> linear_search([0, 5, 7, 10, 15], 0)
+    0
+
+    >>> linear_search([0, 5, 7, 10, 15], 15)
+    4
+
+    >>> linear_search([0, 5, 7, 10, 15], 5)
+    1
+
+    >>> linear_search([0, 5, 7, 10, 15], 6)
+
+    '''
+    if high < low:
+        return None
```

Comment:
Should this raise a ValueError for invalid input?

Reply:
This is the condition that tells us that high has crossed low and the element is not in the list

Reply:
We promised on line 41 that this function would return an it but now we are returning an non-int.  We have two choices:
1. Return -1 as a signal that the element is not in the list --or--
2. Raise an error.

This is like [str.find()](https://docs.python.org/3/library/stdtypes.html#str.find) vs. [str.index](https://docs.python.org/3/library/stdtypes.html#str.index).